src/qibo/tests_new/test_backends_agreement.py

```diff
@@ -0,0 +1,104 @@
+import pytest
+import numpy as np
+from qibo import backends
+from numpy.random import random as rand
+
+
+METHODS = [
+    ("cast", [rand(4)]),
+    ("diag", [rand(4)]),
+    ("reshape", {"x": rand(4), "shape": (2, 2)}),
+    ("stack", [[rand(4), rand(4)]]),
+    ("concatenate", {"x": [rand((2, 3)), rand((2, 4))], "axis": 1}),
+    ("expand_dims", {"x": rand((5, 5)), "axis": 1}),
+    ("copy", [rand(5)]),
+    ("range", {"start": 0, "finish": 10, "step": 2}),
+    ("range", {"start": 0, "finish": 10, "step": 2, "dtype": "DTYPE"}),
+    ("eye", [5]),
+    ("zeros", [(2, 3)]),
+    ("ones", [(2, 3)]),
+    ("zeros_like", [rand((4, 4))]),
+    ("ones_like", [rand((4, 4))]),
+    ("real", [rand(5)]),
+    ("imag", [rand(5)]),
+    ("conj", [rand(5)]),
+    ("mod", [np.random.randint(10), np.random.randint(2, 10)]),
+    ("right_shift", [np.random.randint(10), np.random.randint(10)]),
+    ("exp", [rand(5)]),
+    ("sin", [rand(5)]),
+    ("cos", [rand(5)]),
+    ("pow", {"base": rand(5), "exponent": 4}),
+    ("square", [rand(5)]),
+    ("sqrt", [rand(5)]),
+    ("log", [rand(5)]),
+    ("abs", [rand(5)]),
+    ("expm", [rand((4, 4))]),
+    ("trace", [rand((6, 6))]),
+    ("sum", [rand((4, 4))]),
+    ("sum", {"x": rand((4, 4, 3)), "axis": 1}),
+    ("matmul", [rand((4, 6)), rand((6, 5))]),
+    ("outer", [rand((4,)), rand((3,))]),
+    # ("kron", [rand((4, 4)), rand((5, 5))]),
```

Comment:
Would be great to keep the `kron` and `inv` always on.
Can we add some `pytest.raise` for `tensorflow`?